app/src/main/java/io/intrepid/contest/screens/contestcreation/addcategoriestocontest/AddCategoriesFragment.java

```diff
@@ -0,0 +1,65 @@
+package io.intrepid.contest.screens.contestcreation.addcategoriestocontest;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.OnClick;
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BaseFragment;
+import io.intrepid.contest.base.PresenterConfiguration;
+import io.intrepid.contest.models.Category;
+import io.intrepid.contest.models.Contest;
+import io.intrepid.contest.screens.contestcreation.ContestActivityContract;
+import io.intrepid.contest.screens.contestcreation.ContestCreationFragment;
+import io.intrepid.contest.screens.contestcreation.ValidatableView;
+import timber.log.Timber;
+
+public class AddCategoriesFragment extends BaseFragment<AddCategoriesPresenter> implements AddCategoriesContract.View, ContestCreationFragment {
+    @BindView(R.id.category_name_edittext)
+    EditText categoryNameField;
+    @BindView(R.id.category_description_edittext)
+    EditText categoryDescriptionField;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.fragment_edit_contest_category;
+    }
+
+    @NonNull
+    @Override
+    public AddCategoriesPresenter createPresenter(PresenterConfiguration configuration) {
+        return new AddCategoriesPresenter(this, configuration);
+    }
+
+    @OnClick(R.id.add_new_category_fab)
+    public void onNewCategoryClicked() {
+        categoryNameField.setText("");
+        categoryDescriptionField.setText("");
+        presenter.onCategoryNameEntered(categoryNameField.getText().toString(),
+                                        categoryDescriptionField.getText().toString());
+    }
+
+    @Override
+    public void onNext() {
+        presenter.onCategoriesReady();
```

Comment:
lets rename this onNextClicked, we dont know that the categories are ready based on this firing off alone.